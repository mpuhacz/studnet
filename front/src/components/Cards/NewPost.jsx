import React from 'react';

const NewPost = ({ avatar, name, text, likes }) => (
  <div className="event">
    <div className="label">
      <img src={avatar} alt="avatar"/>
    </div>
    <div className="content">
      <div className="summary">
        <a>{name}</a> zapostowa≈Ç
        <div className="date">
          3 dni temu
        </div>
      </div>
      <div className="extra text">
        {text}
      </div>
      <div className="meta">
        <a className="like">
          <i className="like icon"></i> {likes} Likes
        </a>
      </div>
    </div>
  </div>
);

export default NewPost;
