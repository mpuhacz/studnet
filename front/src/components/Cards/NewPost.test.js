import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NewPost from './NewPost';

describe('<NewPost />', () => {
  it('renders NewPost without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NewPost
      avatar="https://semantic-ui.com/images/avatar/small/justen.jpg"
      name="Name"
      likes={1}
      text="123"
    />, div);
  });

  it('renders NewPost form with correct specification', () => {
    const wrapper = renderer.create(<NewPost
      avatar="https://semantic-ui.com/images/avatar/small/justen.jpg"
      name="Name"
      likes={1}
      text="123"
    />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
