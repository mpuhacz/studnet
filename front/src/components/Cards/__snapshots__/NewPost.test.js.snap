exports[`<NewPost /> renders NewPost form with correct specification 1`] = `
<div
  className="event">
  <div
    className="label">
    <img
      alt="avatar"
      src="https://semantic-ui.com/images/avatar/small/justen.jpg" />
  </div>
  <div
    className="content">
    <div
      className="summary">
      <a>
        Name
      </a>
       zapostowa≈Ç
      <div
        className="date">
        3 dni temu
      </div>
    </div>
    <div
      className="extra text">
      123
    </div>
    <div
      className="meta">
      <a
        className="like">
        <i
          className="like icon" />
         
        1
         Likes
      </a>
    </div>
  </div>
</div>
`;
